### PR TITLE
Move use cases into a separate section and add TPM use case

### PR DESCRIPTION
**Pull request by moidx**
Navigability changes: 

* Move use cases under "OpenTitan > Documentation > Use Cases"
* Move individual use cases into separate subsections.
* Add references to use cases and security sections to the landing page.

Other changes:

* Add TPM use case

### CODE REVIEW

File: doc/use_cases/_index.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+---
+title: "Use Cases"
+---
+
+## Overview
+
+OpenTitan's mission is to raise the security bar industry wide by implementing a
```
Comment (msfschaffner) [inline doc/use_cases/_index.md:7]: nit: should industry-wide be hyphenated?

**Review by msfschaffner — APPROVED** [review]

File: doc/use_cases/_index.md (hunk 1 above)
Comment (moidx) [reply doc/use_cases/_index.md]: Done.

Comment (msfschaffner) [reply doc/use_cases/_index.md]: Thx

**Review by domrizz0 — APPROVED** [review]

File: doc/use_cases/tpm/index.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+## Minimum Crypto Algorithm Requirements
+
+*   TRNG: At least one internal entropy source is required. The entropy source
+    and collector should provide entropy to the state register in a manner that
+    is not visible to an outside process. The entropy collector should regularly
+    update the state register with additional, unbiased entropy.
+*   Hash Algorithms:
+    *   An approved hash algorithm with approximately the same security strength
+        as its strongest asymmetric algorithm. For OpenTitan the target is
+        SHA2-256 (hardware), SHA3-384 (software implementation).
```
Comment (jadephilipoom) [inline doc/use_cases/tpm/index.md:36]: Sorry for the late comment, but wanted to clarify that cryptolib will probably use a **software** SHA2-256 for most cases. TPM sets up hashes with an init/update/final interface that requires the ability to save/restore hash state so multiple processes can hash in an interleaved manner, and the current hardware doesn't have this capability. Also, did you mean SHA3-384 here or SHA2-384? We will have SHA2-384 and SHA2-512 (software) as well as hardware SHA3-{224,256,384,512} from KMAC (but again, not compatible with the init/update/final TPM interface unless only one process is using the SHA3 at a time).

Comment (moidx) [reply doc/use_cases/tpm/index.md:36]: Hi @jadephilipoom, thanks for taking a look at this PR! I created #14537 to fix these issues. Let's continue the discussion there. Thanks

**Pull request merged**